pkg/generator/generator.go

```diff
@@ -39,18 +39,29 @@ type installGenerator struct {
 	providerRegistry registry.ProviderRegistry
 	manifestApi      manifestsapi.ManifestsAPI
 	eventsHandler    eventsapi.Handler
+	installerCache   *installercache.Installers
 }
 
-func New(log logrus.FieldLogger, s3Client s3wrapper.API, cfg Config, workDir string,
-	providerRegistry registry.ProviderRegistry, manifestApi manifestsapi.ManifestsAPI, eventsHandler eventsapi.Handler) *installGenerator {
+type InstallGeneratorDirectoryConfig struct {
```

Comment:
Why create this struct? It doesn't seem to improve anything over just passing the workdir.

Reply:
The original implementation of installercache was instantiated relative to the working directory of the install generator.

I wanted to find a way to leave the responsibility for finding our which root directory would be used by the installer generator with the installer generator.

This had to be done without instantiating the generator as the installercache is a required component of the generator.

TLDR: This is the point

https://github.com/openshift/assisted-service/pull/7205/files#diff-9fde1b602271222d4351f8e635e3398b84f0af0cc0df19ce8b9f5095fefb5c1bR50-R52

Reply:
I see we can use the GeneratorConfig instead, I have moved the things I need to that